mixer/pkg/log/options.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log
+
+import (
+	"github.com/spf13/cobra"
+)
+
+// Options defines the set of options supported by Istio's component logging package.
+type Options struct {
+	// OutputPaths is a list of file system paths to write the log data to.
+	// The special values stdout and stderr can be used to output to the
+	// standard I/O streams.
+	OutputPaths []string
+
+	// JSONEncoding controls whether the log is formatted as JSON.
+	JSONEncoding bool
+
+	// OutputErrors determines whether log messages at the Error level are output.
```

Comment:
curious: why do output leveling in this fashion (vs an explicit Level)?  Is there ever a use-case for output errors and info but not warnings?  If so, how will the code even support that?

This also leads to a proliferation of flags. Wouldn't a single flag for level be simpler?

Reply:
There's no deep motive here, it's just my preference. I rather these actually be treated as flags rather than levels.